src/components/NavScrollItem.js

```diff
@@ -37,79 +35,64 @@ const defaultProps = {
   element: "li",
 }
 
-// A Scroll Aware NavItem compatible with React Bootstrap's Nav
-class NavScrollItem extends React.Component {
-  handleClick = (e) => {
-    const { eventKey, onSelect, onClick, href, history, to } = this.props
+// A Scroll Aware NavItem compatible
+const NavScrollItem = ({
+  active,
+  className,
+  style,
+  activeClass,
+  href,
+  hashSpy,
+  onClick,
+  to,
+  spy,
+  smooth,
+  duration,
+  children,
+  element: Element,
+}) => {
+  const location = useLocation()
+  const root = process.env.GATSBY_BRANCH
+    ? `/${process.env.GATSBY_BRANCH}/`
+    : "/"
 
+  const handleClick = (e) => {
     if (typeof onClick === "function") {
       onClick()
     }
 
     if (href) {
       e.stopPropagation()
 
-      history.push(`${href}#${to || ""}`)
-    }
-
-    if (onSelect) {
-      e.preventDefault()
-
-      onSelect(eventKey, e)
+      window.history.pushState({}, "", `${href}#${to || ""}`)
     }
   }
 
-  render() {
-    const {
-      active,
-      className,
-      style,
-      activeClass,
-      hashSpy,
-      to,
-      spy,
-      smooth,
-      duration,
-      children,
-      element: Element,
-      ...props
-    } = this.props
-
-    delete props.onSelect
-    delete props.eventKey
-
-    // These are injected down by `<Nav>` for building `<SubNav>`s.
-    delete props.activeKey
-    delete props.activeHref
-
-    if (!props.role) {
-      if (props.to === "#") {
-        props.role = "button"
-      }
-    } else if (props.role === "tab") {
-      props["aria-selected"] = active
-    }
-
-    return (
-      <Element
-        role="presentation"
-        className={classNames(className, { active })}
-        style={style}
-      >
-        <Link
+  return (
+    <Element
+      role="presentation"
+      className={classNames(className, { active })}
```

Comment:
Why this `active` class here if we're also including `activeClassName`/`activeClass` below?

Reply:
Good catch! I double check if I can remove this.